crates/sui-indexer/src/handlers/tx_processor.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO remove the dead_code attribute after integration is done
+#![allow(dead_code)]
+
+use async_trait::async_trait;
+use move_binary_format::CompiledModule;
+use move_core_types::language_storage::ModuleId;
+use mysten_metrics::monitored_scope;
+use mysten_metrics::spawn_monitored_task;
+use tokio::sync::watch;
+
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use sui_types::object::Object;
+use tokio::time::Duration;
+use tokio::time::Instant;
+
+use sui_json_rpc::get_balance_changes_from_effect;
+use sui_json_rpc::get_object_changes;
+use sui_json_rpc::ObjectProvider;
+use sui_types::base_types::SequenceNumber;
+use sui_types::digests::TransactionDigest;
+use sui_types::effects::{TransactionEffects, TransactionEffectsAPI};
+use sui_types::transaction::{TransactionData, TransactionDataAPI};
+use tracing::debug;
+
+use sui_types::base_types::ObjectID;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+
+use crate::errors::IndexerError;
+use crate::metrics::IndexerMetrics;
+
+use crate::types_v2::IndexedPackage;
+use crate::types_v2::{IndexedObjectChange, IndexerResult};
+
+/// An in-mem cache for packages. It has static lifetime.
+/// Since we batch process checkpoints, it's possible that when a package
+/// is looked up, it has not been persisted in the database yet. So it
```

Comment:
indexer readers & writer pods are separate, the data sync between reader and writer has to be done via DB sync, can you elaborate the usecase of `lookup before DB commit`, both for package and object?

Reply:
It's used for creating dynamic fields in the write path only. See more [here](https://github.com/MystenLabs/sui/pull/13805#issuecomment-1724405590)